### ISSUE TITLE
Styled element does not work inside ConnectedRouter.

### ISSUE THREAD

**Issue by artur-13** [styled-components/styled-components#2060]
Styled-element used like HOC(for example simple div), does not work inside ConnectedRouter. Simple `div` works great. Please check little example [https://codesandbox.io/embed/lr1j37wl6z](https://codesandbox.io/embed/lr1j37wl6z) . Pages do not switch

**Comment by 0xF013** [comment]
I am pretty sure something is blocking a rerender akin to this https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/blocked-updates.md#dealing-with-update-blocking. Wrapping the container definition like this
```
const Container = withRouter(styled.div`
  padding: 100px;
  font-size: 20px;
`);
```
will fix your problem. Whatever this behaviour is indended I'll leave to people who know more about SC internals.

**Comment by probablyup** [comment]
I actually reverted the PureComponent change in the beta branch but haven't cut a new beta yet

**Comment by probablyup** [comment]
Try beta.10: https://github.com/styled-components/styled-components/releases/tag/v4.0.0-beta.10

**Comment by artur-13** [comment]
Fixed in `v4.0.0-beta.10`. Thanks for the quick response.
